megamek/src/megamek/common/Mech.java

```diff
@@ -1971,7 +1971,8 @@ public HitData rollHitLocation(int table, int side, int aimedLocation, AimingMod
                 // normal front hits
                 switch (roll) {
                     case 2:
-                        if ((getCrew().hasEdgeRemaining()
+                        if (game.getOptions().booleanOption(OptionsConstants.EDGE)
+                            && (getCrew().hasEdgeRemaining()
```

Comment:
It looks like we just have a bunch of this conditional repeated throughout Mech.java (and its inheritors), just with different boolean options.
Would it be viable to add, e.g.

```
public boolean shouldUseEdge(String option) {
    return (
                      game.getOptions().booleanOption(OptionsConstants.EDGE)
                                && getCrew().hasEdgeRemaining()
                                && getCrew().getOptions().booleanOption(option)
                );
```
or something?  Then each check becomes something like this:

```
    case 2:
        if (shouldUseEdge(OptionsConstants.EDGE_WHEN_TAC)
        && !game.getOptions().booleanOption(OptionsConstants.ADVCOMBAT_NO_TAC))
```

Which is somewhat cleaner when spread across all these cases.

Reply:
Would it make sense to implement that method in the `Entity` base class all the various Mech and Aero units extend? Looks like it has a crew member so I think it should have all the information it needs

Reply:
I think that's acceptable; the downside is you would also have to check that `getCrew()` doesn't return `null` prior to calling `.hasEdgeRemaining()` or `.getOptions()`, because while an Entity subclass like Mech _must_ have crew, there's no guarantee that a bare Entity will.